typings/index.d.ts

```diff
@@ -0,0 +1,94 @@
+declare module '@paralleldrive/react-feature-toggles' {
+  import * as React from 'react';
+
+  function Feature(
+    inactiveComponent: any,
+    activeComponent: any,
+    name: string,
+    children?: (args: React.ReactNode) => React.ReactNode
+  ): JSX.Element;
+  export { Feature };
+
+  function FeatureToggles({
+    features
+  }?: {
+    readonly features?: ReadonlyArray<string>;
+    readonly children?: React.ReactNode;
+  }): JSX.Element;
+  export { FeatureToggles };
+
+  interface Feature {
+    readonly name: string;
+    readonly isActive: boolean;
+    readonly dependencies?: ReadonlyArray<string>;
+  }
+
+  function getActiveFeatures(
+    features: ReadonlyArray<Feature>
+  ): ReadonlyArray<string>;
+  export { getActiveFeatures };
+
+  interface Query {
+    readonly ft: string;
+  }
+  function getQueryFeatures(query?: Query): ReadonlyArray<string>;
+  export { getQueryFeatures };
+
+  function mergeFeatures(
+    currentFeatures: ReadonlyArray<string>, // tslint:disable-next-line:readonly-array
+    ...restOfFeatures: Array<ReadonlyArray<string>>
+  ): ReadonlyArray<string>;
+  export { mergeFeatures };
+
+  function removeFeatures(
+    currentFeatures: ReadonlyArray<string>,
+    features: ReadonlyArray<string>
+  ): ReadonlyArray<string>;
+  export { removeFeatures };
+
+  function isActive(
+    name: string,
+    currentFeatures: ReadonlyArray<string>
+  ): boolean;
+  function isActive(
+    name: string
+  ): (currentFeatures: ReadonlyArray<string>) => boolean;
+  export { isActive };
+
+  interface Req {
+    readonly query: Query;
+  }
+
+  function getReqQueryFeatures(req?: Req): ReadonlyArray<string>;
+  export { getReqQueryFeatures };
+
+  function getBrowserQueryFeatures(search?: string): ReadonlyArray<string>;
+  export { getBrowserQueryFeatures };
+
+  interface CurrentActiveFeatures {
+    readonly initialFeatures: ReadonlyArray<string>;
+    readonly req?: Req;
+    readonly search?: string;
+  }
+  function getCurrentActiveFeatures(
+    obj: CurrentActiveFeatures
+  ): ReadonlyArray<string>;
+  export { getCurrentActiveFeatures };
+
+  function configureFeature(
+    inactiveComponent: any
+  ): (name: string) => (activeComponent: any) => () => JSX.Element;
+  function configureFeature(
```

Comment:
This function overload doesn't work. I think it is because the signature above starts out the same and is being seen first. So with this implementation is will work with 
 - `configureFeature(Main)('foo')(Test)`
 - `configureFeature(Main, 'foo')(Test)`
 - `configureFeature(Main, 'foo', Test)`

But will not work with:
 - `configureFeature(Main)('foo', Test)`

Reply:
Hmm let me know if you figure this out.

Does order matter, maybe its the cause?

Have to checked to validate the function does work in this format `configureFeature(Main)('foo', Test)` just to rule out that its not an issue with the curry function from ramda.

Reply:
Yeah, if I flip the order of the top 2 overloads then the opposite works. 

It would then be `configureFeature(Main)('foo', Test)` that worked and `configureFeature(Main)('foo')(Test)` that didn't. 

It does work in that format whenever I flip the order of the overloads, so I that should rule that out right?

Reply:
I think so. Have you tried putting them in this order, this is the order Id expect it to require if the TypeScript system grabs the first overload that matches

- `configureFeature(Main)('foo')(Test)`
- `configureFeature(Main)('foo', Test)`
- `configureFeature(Main, 'foo')(Test)`
- `configureFeature(Main, 'foo', Test)`

Reply:
If I'm not mistaken that is actually the order they are in right now, but I've tried it in every order.

Reply:
Looks like it is :D, I didn't actually look at the file order haha

Reply:
Ok it is working now with all 4 configurations.